ui/src/components/WebRTCVideo.tsx

```diff
@@ -222,10 +224,23 @@ export default function WebRTCVideo() {
       if (settings.mouseMode !== "relative") return;
       // if we ignore the event, double-click will not work
       // if (x === 0 && y === 0 && buttons === 0) return;
-      send("relMouseReport", { dx: calcDelta(x), dy: calcDelta(y), buttons });
+      const dx = calcDelta(x);
+      const dy = calcDelta(y);
+      if (rpcHidReady) {
```

Comment:
I’d suggest not using `useHidRPC()` directly for mouse and keyboard events. Instead, let hidRPC serve purely as a transport layer, with dedicated hooks handling input logic.

We use `useKeyboard()` for keyboard input and `useRPC()` for mouse reports. Mixing approaches like this adds confusion and clutters an already complex HID JS API surface.

A clearer approach would be to introduce both `useMouse()` and `useKeyboard()` hooks. That way, consumption is consistent and obvious, while each hook can encapsulate its own complexity. 

We added `useKeyboard()` out of necessity because its fallback logic was big. That isn’t the case for the mouse, where the fallback is rather simple -- but, I would say the added clarity and consistency still make a `useMouse()` hook worthwhile.

Reply:
Good point, moving mouse logic to `useMouse`